src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2023,6 +2023,52 @@ class NodeUtils(object):
                 gpus[gpuid]['gis'] = {}
             gpus[gpuid]['gis'][giid] = gi
 
+        # Find all MIG UUIDs using nvidia-smi -L
+        # and update it later while finding the CIs
+        uuid_list = []
+        cmd = [self.cfg['nvidia-smi'], '-L']
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'NVIDIA SMI command: %s' % cmd)
+        time_start = time.time()
+        out = []
+
+        try:
+            # Try running the nvidia-smi command
+            process = subprocess.Popen(cmd, shell=False,
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE,
+                                       universal_newlines=True)
+            out = process.communicate()[0].split('\n')
+        except Exception:
+            pbs.logmsg(pbs.EVENT_DEBUG4, 'Failed to execute: %s' %
+                       " ".join(cmd))
+            pbs.logmsg(pbs.EVENT_DEBUG3, '%s: No MIGs found' % caller_name())
+            return
+        elapsed_time = time.time() - time_start
+        if elapsed_time > 2.0:
+            pbs.logmsg(pbs.EVENT_DEBUG,
+                       '%s: nvidia-smi call took %f seconds' %
+                       (caller_name(), elapsed_time))
+
+        # format is:
+        # GPU gpuID: gpu_name (UUID: GPU UUID)
+        # MIG xg.ygb      Device  0: (UUID: MIG-UUID)
+        # MIG xg.ygb      Device  1: (UUID: MIG-UUID)
+
+        line_count = 0
+        for out_line in out:
+            # Don't capture the first list element which has the GPU UUID and
+            # the last line, which has only a newline in it
+            if(line_count == 0 or line_count == len(out)-1):
+                line_count += 1
+                continue
+
+            else:
+                mig_uuid = out_line.split()[-1].rstrip(")")
```

Comment:
Do you also need to strip any '(' ?

Reply:
@Abhishek262 : I think understand since in your design doc, it shows the output that you're parsing:
```
GPU 0: NVIDIA A100-PCIE-40GB (UUID: GPU-ade9b969-b95f-7fc1-9075-860078a3b0b7)
  MIG 1g.5gb      Device  0: (UUID: MIG-1587894d-e0db-5f61-8b47-9aac9ed49baf)
  MIG 1g.5gb      Device  1: (UUID: MIG-b77baeb1-8280-5af4-be1c-0ad656409721)
  MIG 1g.5gb      Device  2: (UUID: MIG-d8b08635-c7ba-55aa-88ac-f86867aad854)
  MIG 1g.5gb      Device  3: (UUID: MIG-98782be8-73f5-5cb4-a225-02158827e203)
  MIG 1g.5gb      Device  4: (UUID: MIG-baeafc17-9af0-5df8-be81-36ee538070d9)
  MIG 1g.5gb      Device  5: (UUID: MIG-3f78d2a1-636f-5a9c-ade4-268bb246fbc0)
  MIG 1g.5gb      Device  6: (UUID: MIG-86c5989a-adb4-528d-ad03-c1f85b86c222)

Reply:
Yeah, that's the case. After splitting, the last element would be `MIG-1587894d-e0db-5f61-8b47-9aac9ed49baf)` which has an extra ')'. Stripping '(' would not be required.